TransformAST.cpp

```diff
@@ -0,0 +1,142 @@
+#include <string>
+
+#include <swift/AST/ASTContext.h>
+#include <swift/AST/ASTWalker.h>
+#include <swift/AST/Expr.h>
+#include <swift/AST/ParameterList.h>
+#include <swift/AST/Pattern.h>
+#include <swift/AST/Stmt.h>
+
+#include "TransformAST.h"
+
+void TransformFinalExpressionAndAddGlobal(swift::SourceFile &src_file)
+{
+    if(!src_file.Decls.empty())
+    {
+        swift::Decl *last_decl = *(src_file.Decls.end() - 1);
+        if(auto last_top_level_code_decl = llvm::dyn_cast<swift::TopLevelCodeDecl>(last_decl))
```

Comment:
Could last_decl ever be a nullptr? If so, dyn_cast isn't going to like that. Maybe `dyn_cast_or_null` would be more appropriate? If you're confident that it should never be nullptr you could add an assertion as well.